### PR TITLE
Bug105

### PR DESCRIPTION
**Pull request by nickdesaulniers**
review? @reqshark 

can you try to test this on OSX? I'm running into an assertion failure on test/bind.js.  Looking into it now.

### CODE REVIEW

**Comment by nickdesaulniers** [comment]
so the failure I was seeing was that there's a file left behind from test runs called some_address, which you can see in test/bind.js.  It looks like if the tests fail, then the file is left behind in a state that makes subsequent tests runs fail assertions.  I get the feeling `flock (2)` is called on this file, and it's never unlocked.  We should unconditionally remove this file after test runs.

**Comment by reqshark** [comment]
I'll take a look... btw nice websocket timeout

**Comment by nickdesaulniers** [comment]
> btw nice websocket timeout

arg!!!

**Comment by nickdesaulniers** [comment]
also, do you think we should remove iojs from travis tests?

**Comment by reqshark** [comment]
I think iojs is a legit version of node, but we could remove it if you want. I use node 4 now so I dont really care about testing it anymore

**Comment by nickdesaulniers** [comment]
while the WS stuff did time out, the test I added passed: `multiple req sockets should not throw` https://travis-ci.org/nickdesaulniers/node-nanomsg/jobs/87511766.  That's not indicative of me not breaking anything though...

**Comment by reqshark** [comment]
that's why skip() would have fired and it would have been appropriate

File: lib/index.js [hunk 1]
```diff
@@ -162,7 +162,7 @@ Socket.prototype._protectArray = function (ret, unwind) {
 Socket.prototype._send = function (msg, flags) {
     if (this.closed) return;
 
-    if(this.type == 'surveyor') {
+    if(this.type == 'surveyor' || this.type == 'rep') {
```
Comment (reqshark) [inline lib/index.js]: i think we should use trip equals

File: lib/index.js [hunk 2]
```diff
@@ -186,6 +186,8 @@ Socket.prototype._receive = function () {
             this.emit('survey-timeout');
             return null;
         }
+    } else if(this.type == 'rep') {
```
Comment (reqshark) [inline lib/index.js]: ditto

**Comment by reqshark** [comment]
looks like it's passing on osx 10.11

**Pull request merged**